pkg/registry/apis/identity/register.go

```diff
@@ -78,53 +85,45 @@ func (b *IdentityAPIBuilder) GetAPIGroupInfo(
 	optsGetter generic.RESTOptionsGetter,
 	dualWriteBuilder grafanarest.DualWriteBuilder,
 ) (*genericapiserver.APIGroupInfo, error) {
-	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(identity.GROUP, scheme, metav1.ParameterCodec, codecs)
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(identityv0.GROUP, scheme, metav1.ParameterCodec, codecs)
 	storage := map[string]rest.Storage{}
 
-	team := identity.TeamResourceInfo
-	teamStore := &legacyTeamStorage{
-		service:        b.Store,
-		resourceInfo:   team,
-		tableConverter: team.TableConverter(),
-	}
-	storage[team.StoragePath()] = teamStore
+	teamResource := identityv0.TeamResourceInfo
+	storage[teamResource.StoragePath()] = team.NewLegacyStore(b.store)
 
-	user := identity.UserResourceInfo
-	userStore := &legacyUserStorage{
-		service:        b.Store,
-		resourceInfo:   user,
-		tableConverter: user.TableConverter(),
-	}
-	storage[user.StoragePath()] = userStore
-	storage[user.StoragePath("teams")] = newUserTeamsREST(b.Store)
-
-	sa := identity.ServiceAccountResourceInfo
-	saStore := &legacyServiceAccountStorage{
-		service:        b.Store,
-		resourceInfo:   sa,
-		tableConverter: sa.TableConverter(),
+	userResource := identityv0.UserResourceInfo
+	storage[userResource.StoragePath()] = user.NewLegacyStore(b.store)
+	storage[userResource.StoragePath("teams")] = team.NewLegacyUserTeamsStore(b.store)
+
+	serviceaccountResource := identityv0.ServiceAccountResourceInfo
+	storage[serviceaccountResource.StoragePath()] = serviceaccount.NewLegacyStore(b.store)
+
+	if b.ssoEnabled {
+		ssoResource := identityv0.SSOSettingResourceInfo
+		storage[ssoResource.StoragePath()] = sso.NewLegacyStore(b.ssoService)
 	}
-	storage[sa.StoragePath()] = saStore
 
 	// The display endpoint -- NOTE, this uses a rewrite hack to allow requests without a name parameter
-	storage["display"] = newDisplayREST(b.Store)
+	storage["display"] = user.NewLegacyDisplayStore(b.store)
 
-	apiGroupInfo.VersionedResourcesStorageMap[identity.VERSION] = storage
+	apiGroupInfo.VersionedResourcesStorageMap[identityv0.VERSION] = storage
 	return &apiGroupInfo, nil
 }
 
 func (b *IdentityAPIBuilder) GetOpenAPIDefinitions() common.GetOpenAPIDefinitions {
-	return identity.GetOpenAPIDefinitions
+	return identityv0.GetOpenAPIDefinitions
 }
 
 func (b *IdentityAPIBuilder) GetAPIRoutes() *builder.APIRoutes {
-	return nil // no custom API routes
+	// no custom API routes
+	return nil
 }
 
 func (b *IdentityAPIBuilder) GetAuthorizer() authorizer.Authorizer {
+	// TODO: handle authorization based in entity.
 	return authorizer.AuthorizerFunc(
 		func(ctx context.Context, a authorizer.Attributes) (authorizer.Decision, string, error) {
-			user, err := identityapi.GetRequester(ctx)
+			user, err := identity.GetRequester(ctx)
```

Comment:
```suggestion
			user, err := claims.From(ctx)
```
🤔  -- that would be better and means we don't have to alias `identityv0`

Reply:
I used `identityv0` because it seems to be a convention in k8s style code. But I can change to this if you prefer it

Reply:
if we could use "identity" I would use that -- but with two imports that want "identity" one has to pick something else -- "identityv0" is totally fine for that.

using just claims, is there a way for us to reasonably block access?

Reply:
No we don't store that information on there.. we would either have to store that information on id tokens and update interface or properly do authorization using authz client that @drclau and @gamab is working on.